### ISSUE TITLE
Add public transport routes

### ISSUE THREAD

**Issue by Robinlovelace** [ITSLeeds/transportData#1]
Heads-up @mem48, I've just added walk, bike and car routes in the releases (not as data as I'd like to keep the package small for CRAN). Do you know a quick and easy way to add a `routes_leeds_pt` dataset?

**Comment by mem48** [comment]
I'll give it a go next week.

If you want to keep the package small on CRAN look at `opentripplanner::otp_dl_jar` I added a cache feature which downloads the files and saves them in the package, so users don't have to download large files each time.

https://github.com/ropensci/opentripplanner/blob/451d33d2f6a1c9d7fd2c86dabff0a7f2a0c7ef64/R/otp-download.R#L22

You could then make a function that downloads the data the first time the package is loaded.

**Comment by Robinlovelace** [comment]
Good idea. I think that providing a small sample of routes, e.g. `routes_leeds_walk_30` for the top 30 routes, and adding something like this would be the best of both worlds, building on the equivalent code in [`stats19`](https://github.com/ropensci/stats19/blob/master/R/stats19-package.R):

```r
.onAttach = function(libname, pkgname) {
  check_if_routes_data_downloaded_ask_if_not()
  msg = paste0(
      "Data provided under OGL v3.0. Cite the source and link to:\n",
      "www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
    )
  packageStartupMessage(msg)
}
```

**Comment by mem48** [comment]
The only issue I can think of is CRAN check not allowing documentation and lazy loading of datasets that are not in the package, so you migh have do make a `routes_leeds_walk_all` object that is the same as `routes_leeds_walk_30` on CRAN and gets replaced with a large object when you call `get_big_data` I'm not sure how well R packages handle being overwritten while loaded. So we might have to test some different options.

Another roption would be to have a small CRAN package `transportdata` that suggests a large github package `transportbigdata`

**Comment by Robinlovelace** [comment]
Both good options. I think using different names for different datasets is the way forward. I can ask Andy South of rnaturalearth how he handled it (with 2 other bigger packages), should chat to him next week so will try to remember.

**Comment by mem48** [comment]
@Robinlovelace Hit a problem for some reason this release of the Transxhcange files is missing data for Leeds, I can do the rail but no busses.

**Comment by Robinlovelace** [comment]
OK. Other options: Google or another commercial service with PT support, if there are any? Is transportAPI still alive?

**Comment by mem48** [comment]
I've pushed at Leeds Rail GTFS file. GTFS files could be another useful data type for the package. I'll have a look at getting some routes, but It might have to wait until I have more time

**Comment by Robinlovelace** [comment]
Great idea, fully agree re GTFS, that is awesome.
